File: mpi/solution/hello.f90
program simple

  implicit none
  ! required MPI include file
  include 'mpif.h'

  integer numtasks, rank, len, ierr
  character(MPI_MAX_PROCESSOR_NAME) hostname

  ! initialize MPI
  call MPI_INIT(ierr)

  ! get number of tasks
  call MPI_COMM_SIZE(MPI_COMM_WORLD, numtasks, ierr)

  ! get my rank
  call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)

  ! get hostanme
  call MPI_GET_PROCESSOR_NAME(hostname, len, ierr)

  ! print task number, rank and hostname
  print '(a,i2,a,i2,a,a)', 'Number of tasks=',numtasks,' My rank=',rank,&
       ' Running on ',hostname

  ! done with MPI
  call MPI_FINALIZE(ierr)

end program simple

File: mpi/solution/hello.c
// required MPI include file
#include "mpi.h"
#include <stdio.h>

int main(int argc, char *argv[]) {
  int  numtasks, rank, len, rc;
  char hostname[MPI_MAX_PROCESSOR_NAME];

  // initialize MPI
  MPI_Init(&argc,&argv);

  // get number of tasks
  MPI_Comm_size(MPI_COMM_WORLD,&numtasks);

  // get my rank
  MPI_Comm_rank(MPI_COMM_WORLD,&rank);

  // get hostname
  MPI_Get_processor_name(hostname, &len);
  // print task number, rank and hostname
  printf ("Number of tasks= %d My rank= %d Running on %s\n", numtasks,rank,hostname);

  // done with MPI
  MPI_Finalize();
}
